### DISCUSSION TITLE
Possible to avoid building the engine for every new game project?

### DISCUSSION THREAD

**Discussion by ohmaya** [o3de/o3de#17354] [Help Wanted]
The engine is supposed to be dynamically linked.

**Comment by VickyAtAZ** [comment]
You don't need to. The you can build the engine Editor and other tools only once. And it can be used for every new game project.

**Reply by ohmaya** [reply]
How?

https://www.docs.o3de.org/docs/welcome-guide/create/creating-projects-using-cli/creating-windows/#create-a-new-o3de-project
https://www.docs.o3de.org/docs/welcome-guide/create/creating-projects-using-cli/creating-windows/#create-a-visual-studio-project
- In Visual Studio, you will build and run the Editor project, which builds both the engine (even if the engine was built previously) and the game projects, and outputs the engine's binaries to <game_project>\build\windows\bin\.

With Unreal Engine, you first build the engine, then start the engine to create a game project, in the Visual Studio solution, there are projects for the game and the engine, you will launch the game project without rebuilding the engine.

**Reply by Andre-LA** [reply]
On Windows by using the installer I can open the project without rebuilding the whole engine, but I'm not a Windows user so I can't tell how far this goes.

But I use on Linux, and by following the Distributable guide, I do also only compile the engine once:
https://docs.o3de.org/docs/user-guide/build/distributable-engine/
